src/main/webapp/app/assessment-shared/assessment-detail/assessment-detail.component.ts

```diff
@@ -26,7 +26,7 @@ export class TextAssessmentDetailComponent {
     }
 
     public delete() {
-        const referencedText = this.block ? this.block.text : this.assessment.reference;
+        const referencedText = this.block ? this.block.text : this.assessment.reference ? this.assessment.reference : '';
```

Comment:
Nit: I try to avoid nested if else, I would propose using boolean logic here:
`const referencedText = this.block && this.block.text || this.assessment.reference || '';`

In my opinion this is clearer when it comes to the different alternatives.

Reply:
I changed it to check if `referencedText` is null when I build `confirmationMessage`. So now it is not needed here